corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js

```diff
@@ -958,6 +947,17 @@ hqDefine("cloudcare/js/form_entry/entries", function () {
             self.question.onchange();
         }
     };
+    FileEntry.prototype.onClear = function () {
+        var self = this;
+        if (self.cleared) {
+            return;
+        }
+        self.cleared = true;
+        self.file(null);
+        self.rawAnswer(constants.NO_ANSWER);
+        self.xformAction = constants.CLEAR_ANSWER;
```

Comment:
in relation to this you'll need to update the [updateXformAction](https://github.com/dimagi/commcare-hq/blob/2bb5b39139e348c73dfd314720b7af0760e6def2/corehq/apps/cloudcare/static/cloudcare/js/form_entry/web_form_session.js#L366) method to account for Signature entries template type.

Reply:
Thank you for catching. What does that do exactly?

Reply:
When you clear an answer the `xformAction` sent to Formplayer is `clear_answer`. Once that request has succeeded it's changing the `xformAction` on the question object back to the appropriate one relative to the template type, so `answer_media` for File entries and `answer` otherwise.

Reply:
For additional context, `clear_answer`, `answer` and `answer_media` are different Formplayer endpoints doing different things with requests.